### ISSUE TITLE
Issue 103: model prediction / assignment を batch 化する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#128]
# Issue 103: model prediction / assignment を batch 化する

## Priority

High

## 背景

現状の `_assign_model_prediction()` は trace ごとに `_predict_model_sec()` を呼ぶ。

```python
pred = trend_model.predict(torch.tensor([float(offset_m)], dtype=torch.float32))
```

30 万 trace 規模では、小さい torch tensor の生成、`predict()` 呼び出し、round/clip/array 代入が trace ごとに発生する。

一方で、同じ model/context を使う trace 群は多い。`_predict_model_array_sec()` はすでに存在し、t0 shift 計算では array prediction が使われている。これを assignment 側にも拡張する。

## 目的

同じ trend model と同じ runtime fit source を共有する trace 群をまとめて prediction し、`physical_center_i` などの result arrays に batch 代入する。品質は現行と完全一致を目標にする。

## 対象ファイル

- `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/physics/physical_center.py`
- 必要なら `packages/seisai-engine/src/seisai_engine/pipelines/fbpick/physics/runtime_diagnostics.py`
- tests:
  - `packages/seisai-engine/tests/test_fbpick_physical_center.py`

### 1. batch assignment helper を追加する

例:

```python
def _assign_model_prediction_batch(
    arrays: dict[str, np.ndarray],
    trace_indices: np.ndarray,
    *,
    trend_model,
    diagnostics: tuple[float, float, float, float, float, float, float] | None,
    plan_by_trace: Mapping[int, _ObservationPlan] | _ObservationPlan,
    offset_abs_m: np.ndarray,
    dt: float,
    n_samples: int,
    runtime_fit_source: int,
    t0_shift_sec: float | np.ndarray = 0.0,
    runtime_diagnostics: PhysicalRuntimeDiagnostics | None = None,
) -> np.ndarray:
    ...
```

返り値は prediction が valid だった trace の mask、または invalid trace indices とする。

### 2. `_predict_model_array_sec()` を assignment で使う

`trace_indices` の offsets をまとめて渡す。

```python
offsets = np.asarray(offset_abs_m, dtype=np.float32)[trace_indices]
physical_t_sec = _predict_model_array_sec(trend_model, offsets) + shift
center_i = np.rint(physical_t_sec / dt)
```

### 3. invalid prediction の fallback を維持する

現行 `_assign_model_prediction()` は prediction が non-finite の場合 false を返し、呼び出し側が fallback する。batch 版も trace ごとに invalid を検出し、invalid trace だけ既存 fallback path に渡せるようにする。

### 4. diagnostics 代入を batch 化する

`_assign_model_diagnostics()` は trace ごと代入だが、同じ diagnostics を共有する場合は配列 slice に一括代入する。

対象 arrays:

```text
physical_model_break_offset_m
physical_model_slope_near_s_per_m
physical_model_slope_far_s_per_m
physical_model_velocity_near_m_s
physical_model_velocity_far_m_s
physical_model_resid_p50_ms
physical_model_resid_p90_ms
```

### 5. runtime diagnostics を調整する

`n_prediction_calls` は現状 trace ごとに increment される。batch 化後は以下のどちらかに整理する。

- 互換重視: trace 数分 increment する。
- 実態重視: batch 呼び出し回数を `n_prediction_batches` として新規追加し、`n_prediction_calls` は trace 数のままにする。

推奨は後者。

## 非対象

- fit context scheduler の全面 refactor。これは Issue 104 とする。
- RANSAC fit の変更。
- anchor selection policy の変更。

### correctness

小規模 synthetic test と Arakawa A0 相当 run で、少なくとも以下が exact match すること。

```text
physical_center_i
physical_center_t_sec
fine_center_i
fine_center_t_sec
physical_model_status
physical_model_failure_reason
physical_runtime_fit_source
```

`physical_center_t_sec` は float32 代入のため、exact match が難しい場合は以下を許容するが、その場合は理由を issue comment に記録する。

```text
max_abs_diff <= 1e-6 sec
physical_center_i exact match
```

### runtime

`prediction_sec` と `assignment_sec` が低下すること。A0 相当の Arakawa one-file run で `physics_total_sec` が悪化しないこと。

## テスト案

- 1 trace の batch が既存 `_assign_model_prediction()` と一致する。
- 複数 trace batch が trace ごとの既存 assignment と一致する。
- `t0_shift_sec` が scalar の場合と vector の場合を確認する。
- trend model が non-finite を返した trace だけ fallback 対象になる。
- diagnostics が全 trace に正しく代入される。

## Arakawa 検証

Issue 101/102 とは独立に検証できるが、最終的には併用して比較する。

比較観点:

```text
prediction_sec
assignment_sec
n_prediction_calls
n_prediction_batches
physical_center_i exact match
status counts match
```

## リスク

- scalar prediction と batch prediction で torch / numpy の丸めが微小に変わる可能性がある。
- `np.rint()` の境界で 1 sample ずれる可能性がある。
- invalid prediction の fallback 処理を batch 内で正しく分岐しないと status count が変わる。
